feathr_project/feathr/definition/feature.py

```diff
@@ -30,6 +30,11 @@ def __init__(self,
                  registry_tags: Optional[Dict[str, str]] = None,
                  ):
         FeatureBase.validate_feature_name(name)
+
+        # Validate the feature type
+        if not isinstance(feature_type, FeatureType):
+            raise KeyError(f'Feature type must be a FeatureType class, like INT32, but got {feature_type}')
```

Comment:
Suggest change to {feature_type} is not a valid feature type, please use one of following valid values: blabla (dump all enum values)

Reply:
It's feasible for `KeyType`, which is a `enum`, but cannot be applied to `FeatureType` as it's a base class. 

From my perspective, message looks too long to dump all enum values: 

KeyType: (all of them needs to add `ValueType` as prefix)
    UNSPECIFIED = 0
    BOOL = 1
    INT32 = 2
    INT64 = 3
    FLOAT = 4
    DOUBLE = 5
    STRING = 6
    BYTES = 7

FeatureType:
BOOLEAN = BooleanFeatureType()
INT32 = Int32FeatureType()
INT64 = Int64FeatureType()
FLOAT = FloatFeatureType()
DOUBLE = DoubleFeatureType()
STRING = StringFeatureType()
BYTES = BytesFeatureType()
FLOAT_VECTOR = FloatVectorFeatureType()
INT32_VECTOR = Int32VectorFeatureType()
INT64_VECTOR = Int64VectorFeatureType()
DOUBLE_VECTOR = DoubleVectorFeatureType()

Reply:
Maybe we should use different sets of names for different types, but that'll be a breaking change.

Reply:
> Maybe we should use different sets of names for different types, but that'll be a breaking change.

Agree, that might be ideal but will introduce breaking change. I think we can simply using this check to help users know what's wrong there